dts/arm/rpi_pico/rp2040.dtsi

```diff
@@ -68,6 +57,34 @@
 			#reset-cells = <1>;
 		};
 
+		syscon: clock-controller@40008000 {
+			compatible = "raspberrypi,pico-syscon";
+			reg = <0x40008000 DT_SIZE_K(4)>;
+			#clock-cells = <1>;
+			status = "okay";
+			ref-frequency = <12000000>;
+			sys-frequency = <125000000>;
+			peri-frequency = <125000000>;
+			usb-frequency = <48000000>;
+			adc-frequency = <48000000>;
+			rtc-frequency = <46875>;
+			xtal_clk: xtal-clk {
```

Comment:
Are these still needed?

Reply:
Yes, this are used inside the driver code

Reply:
Okay, I see you're using them as `source_rate`. But in any case, these should be defined at the board level, as they depend on the value of the oscillator on the board (if it has one). What about choosing external clock sources and setting up the PLLs?